src/nodes.coffee

```diff
@@ -2216,9 +2203,6 @@ exports.Assign = class Assign extends Base
         # Check if @variable contains Obj with splats.
         hasSplat = @variable.contains (node) -> node instanceof Obj and node.hasSplat()
         return @compileDestructuring o if not @variable.isAssignable() or @variable.isArray() and hasSplat
```

Comment:
@zdenko can you remove `or @variable.isArray() and hasSplat`? So that now eg
```
[{...a}] = b
```
would compile to `[{...a}] = b` instead of `({...a} = b[0])`?

Reply:
Done.